### DISCUSSION TITLE
Handling of element in shadow dom

### DISCUSSION THREAD

**Discussion by mujamil90** [grafana/xk6-browser#1207] [General]
I am working with load tests for frontend using xk6-browser and facing issue to handle a link inside shadow dom. I can run same code from playwright but its not working with xk6. I am using Page object model and  sharing code here:

```
export class Homepage {
    constructor(page) {
        this.page = page
        this.verifyDeviceLink = page.locator('.messenger-web-component h4.title')

    }
   async verifyDeviceIfRequired(){
     this.verifyDeviceLink.click();
     await Promise.all([this.verifyDeviceLink.click()]);
    } 
```

and here we have code for test case.

```
  const page = browser.newPage();
    const loginpage = new Loginpage(page);
    const homepage = new Homepage(page);
 
    await loginpage.goto();
    await loginpage.doLogin(" ", "");
    await homepage.verifyDeviceIfRequired();
    await Promise.all([page.waitForNavigation()]);
    page.close();
  } 
```

Kindly help me with this issue.

**Comment by inancgumus** [comment]
Please use our community forum for questions and getting help. Thanks.

https://community.grafana.com/c/grafana-k6/k6-browser/79

**Reply by mujamil90** [reply]
Already posted there

https://community.grafana.com/t/handling-of-element-in-shadow-dom/113929
